pilot/pkg/bootstrap/server.go

```diff
@@ -906,6 +906,19 @@ func (s *Server) initRegistryEventHandlers() {
 					Reason: []model.TriggerReason{model.NamespaceUpdate},
 				})
 			})
+			s.environment.GatewayAPIController.RegisterEventHandler(gvk.Secret, func(_ config.Config, secret config.Config, _ model.Event) {
+				s.XDSServer.ConfigUpdate(&model.PushRequest{
+					Full: true,
```

Comment:
Secret triggering full push is going to lead to a major performance regression, I don't think this will be viable